java/http/vertx/java-http-vertx-consumer/src/main/java/ConsumerApp.java

```diff
@@ -3,64 +3,54 @@
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
 
+import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
-import io.jaegertracing.Configuration;
-import io.opentracing.Tracer;
-import io.opentracing.util.GlobalTracer;
-
-import io.vertx.config.ConfigRetriever;
-import io.vertx.config.ConfigRetrieverOptions;
-import io.vertx.config.ConfigStoreOptions;
 import io.vertx.core.Vertx;
-import io.vertx.core.json.JsonObject;
-import org.apache.logging.log4j.Logger;
+import io.vertx.core.VertxOptions;
+import io.vertx.tracing.opentelemetry.OpenTelemetryOptions;
 import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import io.strimzi.common.TracingSystem;
 
 public final class ConsumerApp {
 
     private static final Logger log = LogManager.getLogger(ConsumerApp.class);
-    
+
     private static String deploymentId;
 
     public static void main(String[] args) throws Exception {
-        Vertx vertx = Vertx.vertx();
-
         CountDownLatch messagesReceivedLatch = new CountDownLatch(1);
         CountDownLatch exitLatch = new CountDownLatch(1);
 
-        ConfigStoreOptions envStore = new ConfigStoreOptions()
-                .setType("env")
-                .setConfig(new JsonObject().put("raw-data", true));
-
-        ConfigRetrieverOptions options = new ConfigRetrieverOptions()
-                .addStore(envStore);
-
-        ConfigRetriever retriever = ConfigRetriever.create(vertx, options);
-
-        retriever.getConfig(ar -> {
-            Map<String, Object> envConfig = ar.result().getMap();
-            HttpKafkaConsumerConfig httpKafkaConsumerConfig = HttpKafkaConsumerConfig.fromMap(envConfig);
-
-            HttpKafkaConsumer httpKafkaConsumer = new HttpKafkaConsumer(httpKafkaConsumerConfig, messagesReceivedLatch);
-
-            vertx.deployVerticle(httpKafkaConsumer, done -> {
-                if (done.succeeded()) {
-                    deploymentId = done.result();
-
-                    if (envConfig.get("JAEGER_SERVICE_NAME") != null) {
-                        Tracer tracer = Configuration.fromEnv().getTracer();
-                        GlobalTracer.registerIfAbsent(tracer);
-                    }
-
-                    log.info("HTTP Kafka consumer started successfully");
-                } else {
-                    log.error("Failed to deploy HTTP Kafka consumer", done.cause());
-                    System.exit(1);
-                }
-            });
+        Map<String, Object> objectMap = new HashMap<>();
+        for (Map.Entry<String, String> entry : System.getenv().entrySet()) {
+            objectMap.put(entry.getKey(), entry.getValue());
+        }
+        HttpKafkaConsumerConfig httpKafkaConsumerConfig = HttpKafkaConsumerConfig.fromMap(objectMap);
```

Comment:
I think this part should be changed the same way we have for the plain Java HTTP examples.
There we have just a `fromEnv` method which is in charge to get values from the env vars. There is no need to build such a map.
See here https://github.com/strimzi/client-examples/blob/main/java/http/java-http-consumer/src/main/java/HttpConsumerConfig.java#L52

Reply:
fromMap now replaced with fromEnv